clash_royale_service.py

```diff
@@ -97,3 +129,47 @@ def get_clan_cards_rank(self, clan_tag, card_type_filter='all'):
         member_cards_ranked = self.sort_list_by_card_level(member_cards_ranked)
 
         return clan_info, member_cards_ranked
+    '''
+    River Race Player Activity History Functions
+    '''
+    def clan_river_race_history(self, clan_tag: str, past_weeks: int = 4) -> Dict[str, PlayerActivity]:
+        clan_players_war_history = {}
+        client_clan_info = self.clash_royale_client.get_clan_info(clan_tag)
+        members_list = client_clan_info['memberList']
+        for member in members_list:
+            clan_players_war_history[member['tag']] = PlayerActivity(
+                tag = member['tag'],
+                name = member['name'],
+                role = member['role'],
+                exp_level = member['expLevel'],
+                fame_hist = [None]* past_weeks,
+                boat_attacks_hist = [None]* past_weeks
+            )
+        return self.past_weeks_clan_war(clan_players_war_history, clan_tag, past_weeks)
+
+    def past_weeks_clan_war(self, clan_players_war_history: Dict[str, PlayerActivity], clan_tag: str, past_weeks: int = 4) -> Dict[str, PlayerActivity]:
+        client_race_log_api_response = self.clash_royale_client.get_river_race_log(clan_tag)
+        grouped_clan_wars = client_race_log_api_response['items'][:past_weeks]
+        current_week = 0
+        # First loop splits the clan wars into specific week
+        for clan_war in grouped_clan_wars:
+            # Second loop identifies the clan within the clan war
+            for standing in clan_war['standings']:
+                clan = standing['clan']
+                if clan['tag'] == clan_tag:
+                    # Third Grab all the participants from the list and store it into the output object
+                    for participant in clan['participants']:
+                        if participant['tag'] in clan_players_war_history:
+                            clan_players_war_history[participant['tag']].fame_hist[current_week] = participant['fame']
+                            clan_players_war_history[participant['tag']].boat_attacks_hist[current_week] = participant['boatAttacks']
+                    break
+            current_week += 1
+        '''
+        Computing War active, elder worthy, and average fame
+        '''

```

Comment:
/supernit does this comment really need 3 lines? Could just use # instead hahahaha